circuits/core/manager.py

```diff
@@ -706,6 +706,14 @@ def _dispatcher(self, event, channels, remaining):  # noqa
             if isinstance(event, generate_events) and self._tasks:
                 event.reduce_time_left(TIMEOUT)
 
+            if err is not None:
+                if event.on_error == "abort":
+                    break
+                elif event.on_error == "ignore":
+                    continue
+                else:
+                    continue
```

Comment:
Since strings are hard to handle (prone to typos), I think it would be better to check if `on_error` is in the allowed options and raise otherwise. What about using constants for those strings?